src/mlpack/methods/det/det_main.cpp

```diff
@@ -166,25 +197,112 @@ void mlpackMain()
   // the given file.
   if (CLI::HasParam("test"))
   {
-    arma::mat testData = std::move(CLI::GetParam<arma::mat>("test"));
+    testData = std::move(CLI::GetParam<arma::mat>("test"));
+    if (CLI::HasParam("test_set_estimates"))
+    {
+      // Compute test set densities.
+      Timer::Start("det_test_set_estimation");
+      arma::rowvec testDensities(testData.n_cols);
 
-    // Compute test set densities.
-    Timer::Start("det_test_set_estimation");
-    arma::rowvec testDensities(testData.n_cols);
-    for (size_t i = 0; i < testData.n_cols; i++)
-      testDensities[i] = tree->ComputeValue(testData.unsafe_col(i));
-    Timer::Stop("det_test_set_estimation");
+      for (size_t i = 0; i < testData.n_cols; i++)
+        testDensities[i] = tree->ComputeValue(testData.unsafe_col(i));
+
+      Timer::Stop("det_test_set_estimation");
 
-    if (CLI::HasParam("test_set_estimates"))
       CLI::GetParam<arma::mat>("test_set_estimates") = std::move(testDensities);
+    }
+
+    // Print variable importance.
+    if (CLI::HasParam("vi"))
+    {
+      arma::vec importances;
+      tree->ComputeVariableImportance(importances);
+      CLI::GetParam<arma::mat>("vi") = importances.t();
+    }
   }
 
-  // Print variable importance.
-  if (CLI::HasParam("vi"))
+  if (CLI::HasParam("tag_file"))
   {
-    arma::vec importances;
-    tree->ComputeVariableImportance(importances);
-    CLI::GetParam<arma::mat>("vi") = std::move(importances.t());
+    const arma::mat& estimationData =
+      CLI::HasParam("test") ? testData : trainingData;
```

Comment:
And this one should be indented four not two spaces since it's a continuation of the previous line. :)  I guess the style checker did not pick up on this one.